cmd/kube-apiserver/app/server.go

```diff
@@ -510,13 +514,26 @@ func getRuntimeConfigValue(s *options.APIServer, apiKey string, defaultValue boo
 	return defaultValue
 }
 
-// Parses the given runtime-config and formats it into map[string]ApiGroupVersionOverride
-func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGroupVersionOverride, error) {
+// Parses the given runtime-config and formats it into genericapiserver.APIResourceEnablement
+func parseRuntimeConfig(s *options.APIServer) (genericapiserver.APIResourceEnablement, error) {
+	v1GroupVersionString := "api/v1"
+	extensionsGroupVersionString := "extensions/v1beta1"
+	batchGroupVersionString := "batch/v1"
+	autoscalingGroupVersionString := "autoscaling/v1"
+	versionToEnablementString := map[unversioned.GroupVersion]string{
```

Comment:
Fair enough.

On Tue, Mar 22, 2016 at 4:20 PM, David Eads <notifications@github.com>
wrote:

> In cmd/kube-apiserver/app/server.go
> <https://github.com/kubernetes/kubernetes/pull/23067#discussion_r57086884>
> :
>
> > @@ -510,13 +514,26 @@ func getRuntimeConfigValue(s *options.APIServer, apiKey string, defaultValue boo
> >  	return defaultValue
> >  }
> >
> > -// Parses the given runtime-config and formats it into map[string]ApiGroupVersionOverride
> > -func parseRuntimeConfig(s *options.APIServer) (map[string]genericapiserver.APIGroupVersionOverride, error) {
> > +// Parses the given runtime-config and formats it into genericapiserver.APIResourceEnablement
> > +func parseRuntimeConfig(s *options.APIServer) (genericapiserver.APIResourceEnablement, error) {
> > +	v1GroupVersionString := "api/v1"
> > +	extensionsGroupVersionString := "extensions/v1beta1"
> > +	batchGroupVersionString := "batch/v1"
> > +	autoscalingGroupVersionString := "autoscaling/v1"
> > +	versionToEnablementString := map[unversioned.GroupVersion]string{
>
> Yeah I agree this function is awful and we shouldn't let the awfulness
> escape-- but a sane parsing of the runtime config string seems like
> something that could be done by your thing?
>
> The more reasonable ways I saw to parse this involve changing the behavior
> from an explicit, hard-coded whitelist into a "really parse the user
> provided data" behavior. I'd prefer to do that in a separate pull rather
> than change behavior in this pull. This one is already pretty big.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/23067/files/cdef546dee0c2d222b6ca0cfee281079b5be2ce6#r57086884>
>